ui/modal/modalPublishPreview/view.jsx

```diff
@@ -151,37 +154,48 @@ const ModalPublishPreview = (props: Props) => {
   const txFee = previewResponse ? previewResponse['total_fee'] : null;
   //   $FlowFixMe add outputs[0] etc to PublishResponse type
   const isOptimizeAvail = filePath && filePath !== '' && isVid && ffmpegStatus.available;
-  let modalTitle;
-  if (isStillEditing) {
-    if (livestream) {
-      modalTitle = __('Confirm Update');
-    } else {
-      modalTitle = __('Confirm Edit');
-    }
-  } else if (livestream) {
-    modalTitle = releasesInFuture ? __('Schedule Livestream') : __('Create Livestream');
-  } else {
-    modalTitle = __('Confirm Upload');
-  }
 
-  let confirmBtnText;
-  if (!publishing) {
+  const [modalTitle, setModalTitle] = React.useState('Upload');
+  const [confirmBtnText, setConfirmBtnText] = React.useState('Save');
+  React.useEffect(() => {
     if (isStillEditing) {
-      confirmBtnText = __('Save');
-    } else if (livestream) {
-      confirmBtnText = __('Create');
+      if (livestream || isLivestreamClaim) {
+        setModalTitle(__('Confirm Update'));
+      } else {
+        setModalTitle(__('Confirm Edit'));
+      }
+    } else if (livestream || isLivestreamClaim || remoteFile) {
+      setModalTitle(
+        releasesInFuture
+          ? __('Schedule Livestream')
+          : (!livestream || !isLivestreamClaim) && remoteFile
+          ? __('Publish Replay')
+          : __('Create Livestream')
+      );
+    } else if (isMarkdownPost) {
+      setModalTitle(__('Confirm Post'));
     } else {
-      confirmBtnText = __('Upload');
+      setModalTitle(__('Confirm Upload'));
     }
-  } else {
-    if (isStillEditing) {
-      confirmBtnText = __('Saving');
-    } else if (livestream) {
-      confirmBtnText = __('Creating');
+
+    if (!publishing) {
+      if (isMarkdownPost) {
+        setConfirmBtnText(__('Post'));
+      } else if (livestream || isLivestreamClaim) {
+        setConfirmBtnText(__('Create'));
+      } else {
+        setConfirmBtnText(__('Upload'));
+      }
     } else {
-      confirmBtnText = __('Uploading');
+      if (isMarkdownPost) {
+        setConfirmBtnText(__('Saving'));
+      } else if (livestream || isLivestreamClaim) {
+        setConfirmBtnText(__('Creating'));
+      } else {
+        setConfirmBtnText(__('Uploading'));
+      }
     }
-  }
+  }, [filePath, isMarkdownPost, isLivestreamClaim, remoteFile]);
```

Comment:
Close if you think these values won't change after the modal is open,
Add if you think it might.
<img width="808" alt="image" src="https://user-images.githubusercontent.com/64950861/177802205-3212a2f1-9bbd-4f8e-8953-322c1f4844ce.png">

Reply:
On closer look, what's the reason it was turned into an effect?  If all the depended variables are calculations from props, we don't need explicit states ... it can be variables themselves like in the old code.

Reply:
I think still not quite right -- need to change from state to variable.  Not really supposed to call set-states from the render function itself anyway.

Reply:
Okay, but using constants would require a ridiculous number of condition checks in this case and would make the line almost unreadable. I used var instead. Is that okay?